src/main/java/Model/PolishBoard.java

```diff
@@ -0,0 +1,80 @@
+package Model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class PolishBoard extends Board {
+
+
+    public PolishBoard() {
+        super(10, 4);
+    }
+
+    @Override
+    public PolishBoard clone() {
+        PolishBoard boardClone = new PolishBoard();
+        for (int i = 0; i < this.getBoardSize(); i++) {
+            for (int j = 0; j < this.getBoardSize(); j++) {
+                Field f = new Field(fields[i][j].getX(), fields[i][j].getY());
+                f.setPawn(fields[i][j].getPawn());
+                boardClone.fields[i][j] = f;
+
+            }
+        }
+        return boardClone;
+    }
```

Comment:
To chyba nie jest konieczne, ale warto pomyśleć, czy nie lepiej przenieść clone() do Board w ogólności.